### ISSUE TITLE
Inform blocks of which adjacent blocks triggered an update

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#5982]
## Description
Blocks like Cactus have a costly update check, checking 5 out of 6 adjacent blocks for support conditions. This could be made cheaper if the update function knew which adjacent block(s) caused the update to be fired.

## Justification
Moar performance

**Comment by dktapps** [comment]
In practice this has no benefit for cacti, as demonstrated by a fixed implementation of #6078.

The theory was that we could avoid having the stem of the cactus checking its neighbours when it grows a new block above. This happens (in theory) because the new upper block triggers updates on all of its neighbours.

However, the reality (and overlooked detail) is that the stem will also trigger an update on itself directly due to the age updating in the same instance. This means that we can't avoid these extra checks during cactus growth in farms, rendering the initial purpose of this change moot.

There will still be gains elsewhere, but cactus farms cannot meaningfully benefit from this change.

**Comment by dktapps** [comment]
A bit more detail: For this to benefit cacti, SELF updates would have to be ignored. However, this is not possible since newly grown cacti will actually need to check their neighbour conditions.

This leaves the following solutions:
- Disable setBlock() updates entirely - not ideal because of stuff like observers as previously mentioned
- Differentiate between private and public blockstate properties, i.e. those which have no visual effect and those which do (this would also be useful for avoiding useless network updates). However, this is also potentially problematic for observers.
- Allow firing neighbour updates without self updates. This  would allow avoiding useless cactus updates while still allowing observers to be notified of their state changes. This doesn't break anything that I know about, but would require an extra parameter to setBlock which I'm not super stoked about.

**Comment by dktapps** [comment]
Cactus performance issues were resolved by 7d200247f88a849d471138cb339955d196beb36e.

@ShockedPlot7560 worked on implementing the feature described here in #6078, but it turned out that the reason for the large performance gain was because of performance issues unrelated to the blocks themselves being updated. I ended up deciding not to go ahead with this, since it required a large amount of extra code for a fairly minimal performance improvement (after fixing aforementioned performance issues).
